File: app/src/main/java/com/example/model/MediaItem.kt
package com.example.model

import java.util.UUID

enum class MediaType {
    PHOTO,
    VIDEO
}

data class MediaItem(
    val id: String = UUID.randomUUID().toString(),
    val uri: String = "",
    val type: MediaType = MediaType.PHOTO,
    val thumbnailUri: String? = null
) {
    fun toFirestoreMap(): Map<String, Any?> = mapOf(
        "id" to id,
        "uri" to uri,
        "type" to type.name,
        "thumbnailUri" to thumbnailUri
    )

    companion object {
        fun fromFirestoreMap(map: Map<String, Any?>): MediaItem {
            return MediaItem(
                id = map["id"] as? String ?: UUID.randomUUID().toString(),
                uri = map["uri"] as? String ?: "",
                type = try {
                    MediaType.valueOf(map["type"] as? String ?: MediaType.PHOTO.name)
                } catch (_: Exception) {
                    MediaType.PHOTO
                },
                thumbnailUri = map["thumbnailUri"] as? String
            )
        }
    }
}

File: iosApp/PrendsEtRends/Models/MediaItem.swift
import Foundation

public enum MediaType: String, Codable {
    case photo = "PHOTO"
    case video = "VIDEO"
}

public struct MediaItem: Identifiable, Codable, Equatable {
    public var id: String
    public var uri: String
    public var type: MediaType
    public var thumbnailUri: String?
    
    public init(
        id: String = UUID().uuidString,
        uri: String,
        type: MediaType = .photo,
        thumbnailUri: String? = nil
    ) {
        self.id = id
        self.uri = uri
        self.type = type
        self.thumbnailUri = thumbnailUri
    }
    
    public func toDictionary() -> [String: Any] {
        var dict: [String: Any] = [
            "id": id,
            "uri": uri,
            "type": type.rawValue
        ]
        if let thumb = thumbnailUri { dict["thumbnailUri"] = thumb }
        return dict
    }
    
    public static func fromDictionary(_ dict: [String: Any]) -> MediaItem {
        let id = dict["id"] as? String ?? UUID().uuidString
        let uri = dict["uri"] as? String ?? ""
        let rawType = dict["type"] as? String ?? MediaType.photo.rawValue
        let type = MediaType(rawValue: rawType) ?? .photo
        let thumb = dict["thumbnailUri"] as? String
        return MediaItem(id: id, uri: uri, type: type, thumbnailUri: thumb)
    }
}
